### PR TITLE
Adding some fixer scripts in dist, formating is src/dbus and fix for mtime in podcasts

### PR DESCRIPTION
**Pull request by sobkas**
This fixer scripts are really ugly, but I use them and it's pain to move the commits out of the way for each pr
Formatting in src/dbus, I just couldn't help myself
mtime was broken after change in PodcastDeleter that set an invalid listened date after deleting episode.

### CODE REVIEW

**Comment by sobkas** [comment]
Uh, hello?

**Comment by davidsansome** [comment]
Please check this Python against the style guide: https://google-styleguide.googlecode.com/svn/trunk/pyguide.html
Particularly indentation (spaces vs tabs), imports (import * is bad) and whitespace around operators (foo = bar instead of foo=bar)

File: dist/copyright_force.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	else:
+		return 0
+
+def gen_copyrights(f):
+
+	commits = []
+	data = {}
+        copyrights = []
+        
+	for ln in Popen(["git", "blame", "--incremental", f], stdout=PIPE).stdout:
```
Comment (davidsansome) [inline dist/copyright_force.py]: subprocess.check_output seems better?

File: dist/copyright_force.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		except KeyError:
+			n = (c["author"], c["author-mail"], set())
+			by_author[c["author"]] = n
+
+		# FIXME: Handle time zones properly
+		year = datetime.fromtimestamp(int(c["author-time"])).year
+
+		n[2].add(year)
+
+	for an, a in list(by_author.iteritems()):
```
Comment (davidsansome) [inline dist/copyright_force.py]: I'm finding it hard to follow this, but can you avoid the O(n^2) somehow?

File: dist/copyright_force.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+def change_file(filename):
+        content=[]
+        out=[]
+        extended=0
+        ends=0
+        with open(filename, "r") as fi:
+                content=fi.readlines()
+        with open("dist/copyright.tmpl", "r") as fi:
+                templet=fi.readlines()
```
Comment (davidsansome) [inline dist/copyright_force.py]: s/templet/template/?

File: dist/copyright_force.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+		if x == prev + 1:
+			prev = x
+			continue
+		
+		if prev == start:
+			r.append("%i" % prev)
+		else:
+			r.append("%i-%i" % (start, prev))
+
+		start = x
```
Comment (davidsansome) [inline dist/copyright_force.py]: It's not really "start" then is it...

File: dist/copyright_force.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+		prev = x
+
+	if not prev is None:
```
Comment (davidsansome) [inline dist/copyright_force.py]: minor: I think if x is not None: is preferred.

File: dist/copyright_force.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python
+
+from subprocess import *
+from sys import *
+from os import rename, remove
+from datetime import *
+
+def pretty_years(s):
```
Comment (davidsansome) [inline dist/copyright_force.py]: At a glance I have no idea what this function, or any other function in this file does.  Please add high-level docstrings, or rewrite the code to be more self-explanatory.

File: src/internet/podcasts/podcastepisode.cpp [hunk 7]
```diff
@@ -146,7 +146,16 @@ void PodcastEpisode::InitFromQuery(const QSqlQuery& query) {
   d->duration_secs_ = query.value(6).toInt();
   d->url_ = QUrl::fromEncoded(query.value(7).toByteArray());
   d->listened_ = query.value(8).toBool();
-  d->listened_date_ = QDateTime::fromTime_t(query.value(9).toUInt());
+
+  // After setting QDateTime to invalid state, it's saved into database as time_t,
+  // when this number(4294967295) is read back from database, it creates a valid QDateTime.
+  // So to make it behave consistently, this change is needed.
+  if (query.value(9).toUInt() == 4294967295) {
```
Comment (davidsansome) [inline src/internet/podcasts/podcastepisode.cpp]: Maybe this would be better as a separate PR?  And also use std::numeric_limits<uint32_t>::max() - 1 or something.

**Comment by sobkas** [comment]
Python stuff shouldn't be in this pr, will remove it now(as it's not my code, only got it from http://0pointer.de/public/copyright.py), so if it's only used by me I will keep it private.
But if there is a need for it, I will merge it with dist/copyright.py and fix all the bugs
Also I will use ULONG_MAX

**Comment by sobkas** [comment]
So now it should be fixed

**Comment by sobkas** [comment]
Instead of ULONG_MAX(or UINT_MAX), I have used std::numeric_limits<unsigned int>::max(), more in C++ style
